zero-code-instrumentation/zero-code-instrumentation.gemspec

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+lib = File.expand_path('lib', __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'version'
+
+Gem::Specification.new do |spec|
+  spec.name        = 'zero-code-instrumentation'
+  spec.version     = VERSION
+  spec.authors     = ['OpenTelemetry Authors']
+  spec.email       = ['cncf-opentelemetry-contributors@lists.cncf.io']
+
+  spec.summary     = 'Zero-code instrumentation for opentelemetry ruby'
+  spec.description = 'Zero-code instrumentation for opentelemetry ruby'
+  spec.homepage    = 'https://github.com/open-telemetry/opentelemetry-ruby-contrib'
+  spec.license     = 'Apache-2.0'
+
+  spec.files = Dir.glob('lib/**/*.rb') +
+               Dir.glob('*.md') +
+               ['LICENSE']
+
+  spec.require_paths = ['lib']
+  spec.required_ruby_version = '>= 3.1'
+
+  spec.add_dependency 'opentelemetry-api', '1.4.0'
```

Comment:
Do you want these to always use strict versions vs pessemistic versioning to allow for bug fixes in dependencies without requiring a new version of this gem?

Reply:
I believe we can opt for a non-strict version of the gem. Initially, I was concerned because opentelemetry-operator prefers strict versioning to prevent unexpected failures. However, since the auto-instrumentation image will be built upstream and is expected to remain stable for a while, using a non-strict version should be safe.

Reply:
Is there going to be a process that bumps this version when the API and SDK versions are updated?